src/notebooksyncagent.cpp

```diff
@@ -782,82 +824,193 @@ bool NotebookSyncAgent::applyRemoteChanges()
     return true;
 }
 
-bool NotebookSyncAgent::updateIncidences(const QList<Reader::CalendarResource> &resources)
+bool NotebookSyncAgent::updateIncidence(KCalCore::Incidence::Ptr incidence, const Reader::CalendarResource &resource, bool *criticalError)
 {
     NOTEBOOK_FUNCTION_CALL_TRACE;
 
-    for (int i=0; i<resources.count(); i++) {
-        const Reader::CalendarResource &resource = resources.at(i);
-        KCalCore::Incidence::Ptr newIncidence = resources.at(i).incidence;
-        if (newIncidence.isNull()) {
-            continue;
-        }
-        if (mLocalDeletedUids.contains(newIncidence->uid())) {
-            LOG_DEBUG("Ignore incidence already deleted locally:" << resource.href);
-            continue;
+    if (incidence.isNull()) {
+        return false;
+    }
+    if (mLocalDeletedUids.contains(incidence->uid())) {
+        LOG_DEBUG("Ignore incidence already deleted locally:" << resource.href);
+        return false;
+    }
+
+    // find any existing incidence with this uid
+    QString incidenceIdentifier = QStringLiteral("%1 (%2)")
+            .arg(incidence->uid())
+            .arg(incidence->hasRecurrenceId() ? incidence->recurrenceId().toString() : QStringLiteral("original"));
+    mStorage->load(incidence->uid());
```

Comment:
Does anything load instance with recurrence id? Optional second parameter on this.